Content.Server/Toilet/ToiletSystem.cs

```diff
@@ -129,31 +129,31 @@ private void OnExamine(EntityUid uid, ToiletComponent component, ExaminedEvent a
             if (!Resolve(uid, ref component, ref meta))
                 return SuicideKind.Special;
 
-            if (!Resolve(uid, ref victimMeta))
+            if (!Resolve(victimUid, ref victimMeta))
                 return SuicideKind.Special;
 
             // check that victim even have head
             if (EntityManager.TryGetComponent<SharedBodyComponent>(victimUid, out var body) &&
                 body.HasPartOfType(BodyPartType.Head))
             {
                 var othersMessage = Loc.GetString("toilet-component-suicide-head-message-others",
-                    ("victim",victimMeta.Name),("owner", meta.Name));
-                _popupSystem.PopupEntity(othersMessage, uid, Filter.Pvs(victimUid));
+                    ("victim",victimMeta.EntityName),("owner", meta.EntityName));
+                _popupSystem.PopupEntity(othersMessage, uid, Filter.Pvs(victimUid).RemoveWhereAttachedEntity(puid => puid == victimUid));
 
                 var selfMessage = Loc.GetString("toilet-component-suicide-head-message",
-                    ("owner", meta.Name));
+                    ("owner", meta.EntityName));
                 _popupSystem.PopupEntity(selfMessage, uid, Filter.Entities(victimUid));
 
                 return SuicideKind.Asphyxiation;
             }
             else
             {
                 var othersMessage = Loc.GetString("toilet-component-suicide-message-others",
-                    ("victim", victimMeta.Name),("owner", meta.Name));
-                _popupSystem.PopupEntity(othersMessage, uid, Filter.Pvs(uid));
+                    ("victim", victimMeta.EntityName),("owner", meta.EntityName));
+                _popupSystem.PopupEntity(othersMessage, uid, Filter.Pvs(uid).RemoveWhereAttachedEntity(puid => puid == victimUid));
 
                 var selfMessage = Loc.GetString("toilet-component-suicide-message",
-                    ("owner",meta.Name));
+                    ("owner",meta.EntityName));
```

Comment:
i think you can just pass the uid and not the entityname into loc.getstring. that way it handles gender/plurality properly in fluent